### ISSUE TITLE
Needs updating to Coinbase Advanced

### ISSUE THREAD

**Issue by asdsadsaasd12312132** [David-Woroniuk/Historic_Crypto#17]
Coinbase Pro is shutting down and migrating to Coinbase Advanced. This needs to be updated to use Coinbase Advanced.

**Comment by David-Woroniuk** [comment]
Agreed - I just saw the below [article](https://www.nerdwallet.com/article/investing/coinbase-vs-coinbase-pro)

I have limited time to work on this presently, but happy to merge any PR's which provide similar functionality for the new version.

**Comment by damian-robinson** [comment]
@David-Woroniuk i'm a noob doing classwork and like that this was so easy to use. could u point me to what i'd need to modify within the code in order to update it to work with coinbase advanced? i'd like to give it a shot.

**Comment by InitialSupport** [comment]
We'd be happy to look into this for you.
Kindly file a support ticket here
https://forms.gle/xABEcR5MwDBJAm1c6

**Comment by tripster202** [comment]
This library seems to match up very closely to the Advanced Trade "Get Candles" endpoint. I will do more research in the coming weeks. This is a good place to start.

https://docs.cloud.coinbase.com/advanced-trade-api/reference/retailbrokerageapi_getcandles

**Comment by asherp** [comment]
Not sure if anyone's looking at this but here some sample code

```py
import pandas as pd
import http.client
import json
from datetime import datetime

# Convert datetime to Unix timestamp
start = int(datetime.strptime("2024-09-11 00:00:00", '%Y-%m-%d %H:%M:%S').timestamp())
end = int(datetime.strptime("2024-09-13 00:00:00", '%Y-%m-%d %H:%M:%S').timestamp())

# Establish connection
conn = http.client.HTTPSConnection("api.coinbase.com")
payload = ''
headers = {
    'Content-Type': 'application/json'
}

# Make request
conn.request("GET", f"/api/v3/brokerage/market/products/BTC-USD/candles?start={start}&end={end}&granularity=FIFTEEN_MINUTE&limit=28", payload, headers)
res = conn.getresponse()
data = res.read()

# Decode and parse JSON data
decoded_data = data.decode("utf-8")
parsed_data = json.loads(decoded_data)
df = pd.DataFrame.from_dict(parsed_data['candles'])

```


<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>start</th>
      <th>low</th>
      <th>high</th>
      <th>open</th>
      <th>close</th>
      <th>volume</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1726164900</td>
      <td>58302.48</td>
      <td>58469.57</td>
      <td>58328.29</td>
      <td>58425.29</td>
      <td>130.32612593</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1726164000</td>
      <td>58250.46</td>
      <td>58419.57</td>
      <td>58399.26</td>
      <td>58328.2</td>
      <td>116.9112028</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1726163100</td>
      <td>58224.39</td>
      <td>58428.02</td>
      <td>58226.17</td>
      <td>58395.4</td>
      <td>141.87815135</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1726162200</td>
      <td>58094.04</td>
      <td>58316.75</td>
      <td>58135.94</td>
      <td>58223.87</td>
      <td>128.29996901</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1726161300</td>
      <td>58117.49</td>
      <td>58298</td>
      <td>58138.72</td>
      <td>58142.36</td>
      <td>129.08781992</td>
    </tr>
  </tbody>
</table>

**Comment by macanudo527** [comment]
@David-Woroniuk  I have a fix for this that I monkey patched on to the app I'm developing. I can submit it, but is this module still being actively developed? It looks like it has ghosted.
